packages/calcite-components/src/components/carousel/carousel.tsx

```diff
@@ -154,17 +158,19 @@ export class Carousel
     disconnectLocalized(this);
     disconnectMessages(this);
     this.clearIntervals();
+    this.resizeObserver?.disconnect();
   }
 
   async componentWillLoad(): Promise<void> {
+    this.setMaxItemsToBreakpoint(this.el.clientWidth);
```

Comment:
`clientWidth` might not reflect the proper size at this point since the component and items are not rendered at this point.

Reply:
I had it in `componentDidLoad()` (like it is in Pagination) - but I was running into `The state/prop "maxItems" changed during "componentDidLoad()"` errors... 

@jcfranco can the resize observer be relied upon to fire during component initialization? Because `setMaxItemsToBreakpoint` is run there as well as in one of these lifecycle periods.

Reply:
Yes, the callback runs when the [observer is created](https://codepen.io/jcfranco/pen/NWemRed?editors=1111) and any resize afterwards.

Reply:
In tests as well? (Seems to work .. )